### PR TITLE
Rework logic to prevent calling wrong install function in kpm

### LINKED ISSUE #167: kpm installs linked to a raw coffee script fail to install
`/kpm install xkcd` failed to install with the following message

```
fatal: repository 'https://raw.githubusercontent.com/github/hubot-scripts/master/src/scripts/xkcd.coffee/ ' not found
Cloning into '/tmp/tmp-4472Y2VPkienKvC8'...
fatal: repository 'https://raw.githubusercontent.com/github/hubot-scripts/master/src/scripts/xkcd.coffee/ ' not found
```

But was successfully installed when manually linking the script as

`/kpm install https://raw.githubusercontent.com/github/hubot-scripts/master/src/scripts/xkcd.coffee`

Possible cause is the `/` at the end of the link added by kpm

### PR DESCRIPTION
**Pull request by mrkno**
resolves #167

### CODE REVIEW

File: core/core_modules/kpm.js [hunk 1]
```diff
@@ -287,29 +288,30 @@ var git = require.once('../git.js'),
 
                 for (var i = 0; i < args.length; i++) {
                     var url = args[i];
+                    var spl = url.split('/');
+                    if (spl.length === 1) {
+                        refreshModuleTable(url, function(u, err) {
+                                if (err || !moduleTable.modules[u]) {
+                                    return;
+                                }
+                                url = moduleTable.modules[u];
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: [Expected indentation of 28 space characters but found 32.](https://www.codacy.com/app/theprojectshop/Kassy/file/2719763881/issues/source?bid=3282987&fileBranchId=3347522#l297)

File: core/core_modules/kpm.js [hunk 2]
```diff
@@ -287,29 +288,30 @@ var git = require.once('../git.js'),
...
+                            }.bind(this));
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: [Expected indentation of 24 space characters but found 28.](https://www.codacy.com/app/theprojectshop/Kassy/file/2719763881/issues/source?bid=3282987&fileBranchId=3347522#l298)

File: core/core_modules/kpm.js [hunk 3]
```diff
@@ -287,29 +288,30 @@ var git = require.once('../git.js'),
 
                 for (var i = 0; i < args.length; i++) {
                     var url = args[i];
+                    var spl = url.split('/');
+                    if (spl.length === 1) {
+                        refreshModuleTable(url, function(u, err) {
+                                if (err || !moduleTable.modules[u]) {
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: [Expected indentation of 28 space characters but found 32.](https://www.codacy.com/app/theprojectshop/Kassy/file/2719763881/issues/source?bid=3282987&fileBranchId=3347522#l294)

File: core/core_modules/kpm.js (hunk 1 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: [Expected indentation of 28 space characters but found 32.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018391/issues/source?bid=3282987&fileBranchId=3347522#l297)

File: core/core_modules/kpm.js (hunk 2 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: [Expected indentation of 24 space characters but found 28.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018391/issues/source?bid=3282987&fileBranchId=3347522#l298)

File: core/core_modules/kpm.js (hunk 3 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: [Expected indentation of 28 space characters but found 32.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018391/issues/source?bid=3282987&fileBranchId=3347522#l294)

File: core/install.js [hunk 4]
```diff
@@ -12,11 +12,25 @@
...
 
 load({loglevel: 'silent'});
 var inst = deasync(npm.commands.install),
-upd = deasync(npm.commands.update),
+    upd = deasync(npm.commands.update),
+
+// inject require modifications into all coffeescript code because babel wont
+coffeescriptRequireInjector = function () {
+    if (!csHasLoaded && global.coffeescriptLoaded) {
+        var cs = require("coffee-script"),
```
Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018399/issues/source?bid=3282987&fileBranchId=3347522#l25)

Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018399/issues/source?bid=3282987&fileBranchId=3347522#l25)

File: core/install.js [hunk 5]
```diff
@@ -12,11 +12,25 @@
...
+    load = deasync(npm.load),
+    csHasLoaded = false;
 
 load({loglevel: 'silent'});
 var inst = deasync(npm.commands.install),
-upd = deasync(npm.commands.update),
+    upd = deasync(npm.commands.update),
+
+// inject require modifications into all coffeescript code because babel wont
+coffeescriptRequireInjector = function () {
```
Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018399/issues/source?bid=3282987&fileBranchId=3347522#l23)

File: core/require.js [hunk 6]
```diff
@@ -66,9 +66,12 @@ module.exports = function () {
     return {
         visitor: {
             Program(path) {
-                path.unshiftContainer('body', babylon.parse('require = (global || GLOBAL).requireHook(require);').program.body[0]);
+                path.unshiftContainer("body", babylon.parse(module.exports.injectionString).program.body[0]);
```
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018411/issues/source?bid=3282987&fileBranchId=3347522#l69)

File: core/require.js [hunk 7]
```diff
@@ -66,9 +66,12 @@ module.exports = function () {
...
             }
         }
     };
 };
+
+module.exports.injectionString = "require = (global || GLOBAL).requireHook(require);";
```
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018411/issues/source?bid=3282987&fileBranchId=3347522#l75)

File: core/require.js (hunk 6 above)
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018411/issues/source?bid=3282987&fileBranchId=3347522#l69)

File: core/require.js (hunk 7 above)
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737018411/issues/source?bid=3282987&fileBranchId=3347522#l75)

File: core/require.js (hunk 6 above)
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027424/issues/source?bid=3282987&fileBranchId=3347522#l69)

File: core/require.js (hunk 7 above)
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027424/issues/source?bid=3282987&fileBranchId=3347522#l75)

File: core/require.js [hunk 8]
```diff
@@ -26,7 +26,7 @@ global.requireHook = function (req) {
 
     func.searchCache = function (moduleName, callback) {
         var mod = func.resolve(moduleName);
-        if (mod && (typeof (mod = func.cache[mod]) !== 'undefined')) {
+        if (mod && (typeof (mod = func.cache[mod]) !== "undefined")) {
```
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027424/issues/source?bid=3282987&fileBranchId=3347522#l29)

File: core/require.js [hunk 9]
```diff
@@ -10,10 +10,10 @@
  *		Copyright (c) Matthew Knox and Contributors 2016.
  */
 
-'use strict';
+"use strict";
```
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027424/issues/source?bid=3282987&fileBranchId=3347522#l13)

File: core/require.js [hunk 10]
```diff
@@ -10,10 +10,10 @@
...
 
-var babylon = require('babylon'),
-    inst = require('./install.js');
+var babylon = require("babylon"),
```
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027424/issues/source?bid=3282987&fileBranchId=3347522#l15)

File: core/require.js [hunk 11]
```diff
@@ -10,10 +10,10 @@
...
+    inst = require("./install.js");
```
Comment (mrkno) [inline core/require.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027424/issues/source?bid=3282987&fileBranchId=3347522#l16)

File: core/install.js (hunk 4 above)
Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027412/issues/source?bid=3282987&fileBranchId=3347522#l25)

Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027412/issues/source?bid=3282987&fileBranchId=3347522#l25)

File: core/install.js (hunk 5 above)
Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027412/issues/source?bid=3282987&fileBranchId=3347522#l23)

File: core/install.js (hunk 4 above)
Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027958/issues/source?bid=3282987&fileBranchId=3347522#l25)

Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027958/issues/source?bid=3282987&fileBranchId=3347522#l25)

File: core/install.js (hunk 5 above)
Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737027958/issues/source?bid=3282987&fileBranchId=3347522#l23)

File: core/install.js [hunk 12]
```diff
@@ -12,20 +12,35 @@
...
 
-install = function(name) {
-    console.info('Installing "' + name + '" from npm.');
-    inst([name]);
-    console.info('Installation complete.');
-};
+    // inject require modifications into all coffeescript code because babel wont
+    coffeescriptRequireInjector = function () {
+        if (!csHasLoaded && global.coffeescriptLoaded) {
+            var cs = require("coffee-script"),
```
Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737028527/issues/source?bid=3282987&fileBranchId=3347522#l25)

Comment (mrkno) [inline core/install.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2737028527/issues/source?bid=3282987&fileBranchId=3347522#l25)

**Comment by Awarua-** [comment]
Approved

**Comment by ParasDPain** [comment]
Approved

**Pull request merged**